projects/perception/skeleton_based_action_recognition/demos/demo.py

```diff
@@ -35,7 +35,7 @@ class VideoReader(object):
     def __init__(self, file_name):
         self.file_name = file_name
         try:  # OpenCV needs int to read from webcam
-            self.file_name = int(file_name)
+            self.file_name = file_name
```

Comment:
This try block will always succeed after the change. So, if we do not need this it should be altogether removed. Why is this change needed?

Reply:
because you can bring not only webcams but also other videos. for example, rtsp(cctv) streams.

Reply:
In the modified version of the code the `try` block can be removed altogether, since the assignment has already been done in l.36. Removing lines 37-40 in the current version (without the int cast) will not change anything.

However, I was trying to understand why this block exists in the first place.   To the best of my understanding, the behavior of VideoCapture changes according to the data types it receives (https://docs.opencv.org/3.4/d8/dfe/classcv_1_1VideoCapture.html). When an `int` is passed, then it tries to open the corresponding camera. On the other hand, when a string is passed, then it tries to open the corresponding stream or file. The `try`... `except` block is essentially trying to cope this behavior - in a not very clear way (especially given that the argument is called `file_name`). If the supplied `file_name` string can be cast into an `int`, then it implicitly forces opencv to read from the corresponding webcam. Of course, if there is a file with no suffix and can be cast into a number, then this code will not read from the file and it will try to open the corresponding webcam. However, in any other case it should function correctly. For example, you shouldn't have issues with reading from rtsp streams using the existing version of the code.

Reply:
oh, i see!